influxql/ast_test.go

```diff
@@ -1700,6 +1701,102 @@ func TestParse_Errors(t *testing.T) {
 	}
 }
 
+// This test checks to ensure that we have given thought to the database
+// context required for security checks.  If a new statement is added, this
+// test will fail until it is categorized into the correct bucket below.
+func Test_EnforceHasDefaultDatabase(t *testing.T) {
+	pkg, err := importer.Default().Import("github.com/influxdata/influxdb/influxql")
+	if err != nil {
+		fmt.Printf("error: %s\n", err.Error())
+		return
+	}
+	statements := []string{}
+
+	// this is a list of statements that do not have a database context
+	exemptStatements := []string{
+		"CreateDatabaseStatement",
+		"CreateUserStatement",
+		"DeleteSeriesStatement",
+		"DropDatabaseStatement",
+		"DropMeasurementStatement",
+		"DropSeriesStatement",
```

Comment:
Same here.  If you have multiple sources, DefaultDatabase is the wrong mechanism to use.